--- prober.c.orig	2013-04-30 21:02:47.000000000 +0000
+++ prober.c	2013-04-30 21:23:44.000000000 +0000
@@ -62,6 +62,7 @@
 unsigned int clientip = 0;
 
 unsigned int verbose = 0;
+int use_global_server = 0;
 
 
 /* Utility functions.  */
@@ -219,13 +220,18 @@
   serverip = htonl(str2ip("143.215.129.100"));
   //serverip = htonl(str2ip("38.102.0.111"));
 
-  while ((c = getopt (argc, argv, "vh")) != -1)
+  while ((c = getopt (argc, argv, "vhn:")) != -1)
   {
   switch (c)
   {
   case 'v':
 	  verbose = 1;
 	  break;
+  case 'n':
+      use_global_server = 1;
+      printf("Using: server %s\n", optarg);
+      serverip = htonl(str2ip(optarg));
+      break;
   case '?':
   case ':':
   case 'h':
@@ -321,8 +327,11 @@
 
   sleepRes = prober_sleep_resolution();
 
-  //tcpsock = connect2server(serverip, fileid);
-  tcpsock = selectServer(fileid);
+  if ( 1 == use_global_server ) {
+     tcpsock = connect2server(serverip, fileid);
+  } else {
+     tcpsock = selectServer(fileid);
+  }
   if(tcpsock <= 0) printf("Server busy. Please try again later.\n");
   CHKRET(tcpsock);
 
